packages/oasis-actions/src/operations/aave/deposit.ts

```diff
@@ -0,0 +1,104 @@
+import BigNumber from 'bignumber.js'
+import { assert } from 'console'
+import { aaveDeposit } from '../../actions/aave'
+import { wrapEth, pullToken, swap, setApproval } from '../../actions/common'
+import { ADDRESSES } from '../../helpers/addresses'
+import { OPERATION_NAMES } from '../../helpers/constants'
+
+interface SwapArgs {
+  fee: number
+  receiveAtLeast: BigNumber
+  calldata: string
+  collectFeeInFromToken: boolean
+}
+
+// TODO: Probably put a generic about the tokens i.e DepositArgs<Tokens> and provide AAVETokens
+// TODO: Take into consideration the precision for the tokens. So indeed might be a good idea for the Tokens generics
+interface DepositArgs {
+  // - either for a swap where the `entryToken` will be exchanged for the `depositToken`
+  // - or it will be directly deposited in the protocol
+  entryToken: string
+  // - either used for a swap if `entryToken` is swapped for `depositToken`
+  // - or it will be directly deposited in the protocol
+  amount: BigNumber
+  // - if it's omitted that means that the `entryToken` with bbe used in the deposit
+  // - if it's provided that means that the `entryToken` will be swapped for `depositToken`
+  depositToken?: string
+  // Used to pull tokens from if ERC20 is used in the deposit
+  depositorAddress: string
+  // In order to borrow assets on aave the deposited ones ( lent ) should be allowed to be used as collateral.
+  allowDepositTokenAsCollateral: boolean
+  // Must be provided if `depositToken` is also provided
+  swapArgs?: SwapArgs
+}
+export async function deposit({
+  entryToken,
+  depositToken,
+  amount,
+  depositorAddress,
+  swapArgs,
+}: DepositArgs) {
+  const isSwapNeeded = !!depositToken
```

Comment:
What is your opinion on being more explicit, instead of not passing `depositToken` consumer should always provide both. In this case `const isSwapNeeded = entryToken !== depositToken`?

Reply:
Was thinking about it. The idea is to not add all sort of properties when we can achieve something with the existing ones. If you think it's too confusing / complicating and etc I can agree to just provide a boolean value and that's it :) . Probably being more explicit is better and will lead to less issues / bugs in the future.

Reply:
Another case what if they are both defined but are different, the logic would assume the swap is not necessary thus the tx would fail or the outcome will be unpredictable.

Reply:
On the frontend you don't call `deposit` from `operation` but `deposit` from `strategy`.

Reply:
> Another case what if they are both defined but are different, the logic would assume the swap is not necessary thus the tx would fail or the outcome will be unpredictable.

I don't follow. There are even tests exploring different scenarios where if the two are different there is a swap.